### PR TITLE
add stubs for frozendict

### PR DESCRIPTION
**Pull request by sonlamho**
Requesting to add stubs for this frozendict package: https://pypi.org/project/frozendict/

### CODE REVIEW

**Comment by srittau** [comment]
Thank you. A general remark (not a full review): `frozendict` should derive from `typing.Mapping`, the generic version of `collections.Mapping`, and should be generic itself. This provides more utility, since the keys and values can be type checked.

**Comment by sonlamho** [comment]
> Thank you. A general remark (not a full review): `frozendict` should derive from `typing.Mapping`, the generic version of `collections.Mapping`, and should be generic itself. This provides more utility, since the keys and values can be type checked.

Thanks for your comment. I think I have improved it in that direction

File: third_party/3/frozendict.pyi [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    dict_cls: Type[Dict] = ...
+    @overload
+    def __init__(self, **kwargs: _VT) -> None: ...
+    @overload
+    def __init__(self, mapping: Mapping[_KT, _VT]) -> None: ...
+    @overload
+    def __init__(self, iterable: Iterable[Tuple[_KT, _VT]]) -> None: ...
+    def __getitem__(self, key: _KT) -> _VT: ...
+    def __contains__(self, key: object) -> bool: ...
+    def copy(self: _S, **add_or_replace) -> _S: ...
```
Comment (srittau) [inline third_party/3/frozendict.pyi:19]:
```suggestion
    def copy(self: _S, **add_or_replace: _VT) -> _S: ...
```

Ideally, we'd be able to restrict `_S` to `str` keys here, but I have no good idea how to do that.

**Review by srittau — CHANGES_REQUESTED** [review]
Thank you. One missing type below.

File: third_party/3/frozendict.pyi (hunk 1 above)
Comment (sonlamho) [reply third_party/3/frozendict.pyi]: suggested change applied, thanks!
it's possible that the frozendict has mixed key types, though it's true most of the time it's used with str keys only.

**Review by srittau — APPROVED** [review]

**Pull request merged**